From 8cf9a856b875076761314f7a300c250ba1140a78 Mon Sep 17 00:00:00 2001
From: Kiran Kumar K <kirankumark@marvell.com>
Date: Mon, 2 Nov 2020 15:48:14 +0530
Subject: [PATCH 0834/1921] octeontx2-af: limit KPU parsing for GTPU only to
 GTPv1

While parsing GTPU, all GTP types like GTPv1, GTPv2 and GTP' has been
extracted as single generic ltype as NPC_LT_LE_GTPU. This will create
a problem when matching needs to be performed only for specific versions.
This patch limit the KPU parsing for GTPU only to GTPv1. Other GTP types
like GTPv2 and GTP' won't be parsed as GTPU. If KPU parsing support is
required for GTPv2 or GTP' then different ltypes has to be added.

Signed-off-by: Kiran Kumar K <kirankumark@marvell.com>
Change-Id: I256e505df87b7f5689f73712a5b318768f4aa330
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/39382
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/npc_profile.h    | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index 8a5d5a92bda0..89a53ec7b25b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -6705,8 +6705,8 @@ static struct npc_kpu_profile_cam kpu9_cam_entries[] = {
 		NPC_S_KPU9_GTPU, 0xff,
 		0x0000,
 		0x0000,
-		0x0000,
-		0x0000,
+		NPC_GTP_PT_GTP | NPC_GTP_VER1,
+		NPC_GTP_PT_MASK | NPC_GTP_VER_MASK,
 		0x0000,
 		0x0000,
 	},
@@ -12858,10 +12858,10 @@ static struct npc_kpu_profile_action kpu9_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 1,
-		NPC_S_NA, 0, 1,
+		8, 0, 6, 2, 0,
+		NPC_S_KPU12_TU_IP, 8, 1,
 		NPC_LID_LE, NPC_LT_LE_GTPU,
-		NPC_F_LE_L_GTPU_UNK,
+		0,
 		0, 0, 0, 0,
 	},
 	{
-- 
2.31.1

